eng/pipelines/common/templates/runtimes/run-test-job.yml

```diff
@@ -57,6 +58,11 @@ jobs:
         - '${{ parameters.runtimeFlavor }}_common_test_build_p0_${{ parameters.managedTestBuildOsGroup }}${{ parameters.managedTestBuildOsSubgroup }}_${{ parameters.archType }}_${{parameters.buildConfig }}'
       - ${{ if ne(parameters.testGroup, 'innerloop') }}:
         - '${{ parameters.runtimeFlavor }}_common_test_build_p1_${{ parameters.managedTestBuildOsGroup }}${{ parameters.managedTestBuildOsSubgroup }}_${{ parameters.archType }}_${{parameters.buildConfig }}'
+      - ${{ if eq(parameters.testBuildPhased, true) }}:
```

Comment:
Why do we have to still depend on the non-phased build if this is to true?

Reply:
`testBuildPhased` here means that the tests were built in two phases.  `TargetGeneric` and `TargetSpecific` tests. We join them here by unzipping into the same directory.

Reply:
So if testBuildPhased==true we still depend on: 
`- '${{ parameters.runtimeFlavor }}_common_test_build_p0_${{ parameters.managedTestBuildOsGroup }}${{ parameters.managedTestBuildOsSubgroup }}_${{ parameters.archType }}_${{parameters.buildConfig }}'`

Reply:
Yes

Reply:
@trylek And I discussed this briefly during the original review.  

We may eventually refactor and build the `TargetSpecific` tests as part of `run-test-job`.  He was suggesting collecting stats to determine whether the extra build legs added value or not.

Reply:
That sounds like a reasonable, interesting idea.

Reply:
In fact, I was mostly worried (based on earlier discussions with @jashook) that adding too many legs can be counterproductive by increasing the load on machine allocation and provisioning. I however think that having the specific test build legs initially separate is probably a good idea to get an initial reading on the perf characteristics.

Reply:
Right, but if we refactor and just have build-test legs for the generic tests and then the target specific builds are done at the run-test-job, before sending to helix, we would be reducing the number of legs we will have by the number of target specific tests we currently have, so I think it would improve perf and reliability.

Reply:
I suspect you are right.

+ We have reduce the target specific build by at least 90%.
+ The extra leg appears expensive (maybe 50% overhead). I was seeing target specific 10 minute builds in the pipelines.  I don't think it was that long on my local machine 
+ We should only have to rerun a given leg < 1% of the time (if we achieve 95% success).
+ Fewer legs should reduce infra burden, so we could se benefits elsewhere.
+ Seems like the savings could easily amortize the cost.

Filtering to build the TargetSpecific tests is currently slow.  But if we make msbuild improvements this may go away.